plugins/fishlim/plugin_hexchat.c

```diff
@@ -170,14 +208,14 @@ static int handle_incoming(char *word[], char *word_eol[], hexchat_event_attrs *
     
     /* Try to decrypt with these (the keys are searched for in the key store) */
     encrypted = word[ew+1];
-    decrypted = fish_decrypt_from_nick(recipient, encrypted);
-    if (!decrypted) decrypted = fish_decrypt_from_nick(sender_nick, encrypted);
+    decrypted = fish_decrypt_from_nick(recipient, encrypted, &mode);
+    if (!decrypted) decrypted = fish_decrypt_from_nick(sender_nick, encrypted, &mode);
     
     /* Check for error */
     if (!decrypted) goto decrypt_error;
     
     /* Build unecrypted message */
-    message = g_string_sized_new (100); /* TODO: more accurate estimation of size */
+    message = g_string_sized_new (strlen(word_eol[1])); /* Just put at least the size of original command */
```

Comment:
I'm not sure that is a useful estimation, since you are adding more text it has to realloc sometimes anyway.

Reply:
I refactorized all this for get code sustainable